src/components/helpers/DownloadButton.vue

```diff
@@ -13,30 +14,22 @@ import getResourceURL from '@/helpers/resourceURL';
 
 export default {
   props: {
-    hash: {
-      type: String,
+    file: {
+      type: Object,
       required: true,
     },
-    title: {
-      type: String,
-      required: false,
-      default: 'download',
-    },
   },
   methods: {
-    download() {
-      const rURL = getResourceURL(this.hash);
-
-      fetch(rURL)
-        .then((response) => response.blob())
-        .then((blob) => {
-          const url = window.URL.createObjectURL(blob);
-          const link = document.createElement('a');
-          link.href = url;
-          link.download = this.title;
-          link.click();
-        })
-        .catch(console.error);
+    getResourceURL,
+    /**
+     * strip tags from title
+     * @returns {string}
+     */
+    fileName() {
+      return new DOMParser()
```

Comment:
Wouldn't it be simpler just to use the first reference and the parent hash to construct the URL?
`/ipfs/${parent_hash}/${name}`? That would also make sure we have an extension most cases.

Reply:
This is actually a comment, feel free to keep it as-is or change it as you like.

Reply:
can you point me to some documentation about this? I am not sure what you mean. I am happy to make it simpler, because this tag stripping seems not straightforward.

Reply:
Sure. The [API docs](https://app.swaggerhub.com/apis-docs/ipfs-search/ipfs-search/1.0.2#/default/get_search) give an example result, such as this one:
```json
{
      "hash": "QmcDCte64xtxqTVzdWnT5MG9yi3dFsNuLZjAyess4RJFWc",
      "title": "Marine Combat <em>Water</em> Survival by Anonymous",
      "description": "This publication also teaches Marines to cross <em>water</em> obstacles and perform <em>water</em> rescues correctly and safely.",
      "type": "file",
      "size": 5605,
      "first-seen": "2018-08-18T20:49:17Z",
      "last-seen": "2020-10-27T12:55:32Z",
      "score": 231.37302,
      "references": [
        {
          "parent_hash": "QmcDCte64xtxqTVzdWnT5MG9yi3dFsNuLZjAyess4RJFWc",
          "name": "marine-combat-water-survival.html"
        }
      ],
      "mimetype": "text/html"
}
```

Then you could do something like this (this is loosehand, untested):
```javascript
function getURL(item) {
    let reference = item.references.get(0);
    if (reference) {
        return`/ipfs/${reference.parent_hash}/${name}`;
    }
    // No reference available, resort to plain hash.
    return `/ipfs/${item.hash};
}
```

Reply:
awesome, I will make it. 
But how do these references work, what does it mean? Aren't there sometimes multiple references? Why does the reference have a name of the file?